src/screens/Platform/DAppDisclaimer.js

```diff
@@ -0,0 +1,223 @@
+import LiveIcon from "../../icons/LiveLogo";
+import LText from "../../components/LText";
+import CloseIcon from "../../icons/Close";
+import InfoIcon from "../../icons/Info";
+import { rgba } from "../../colors";
+import AppIcon from "./AppIcon";
+
+export type Props = {
+  closeDisclaimer: () => void,
+  disableDisclaimer: () => void,
+  onContinue: () => void,
+  isOpened: boolean,
+  icon?: string | null,
+};
+
+const DAppDisclaimer = ({
+  closeDisclaimer,
+  isOpened,
+  disableDisclaimer,
+  onContinue: next,
+  icon,
+}: Props) => {
+  const { t } = useTranslation();
+  const { colors } = useTheme();
+  const [
+    disableDisclaimerChecked,
+    setDisableDisclaimerChecked,
+  ] = useState<boolean>(false);
+
+  const onClose = useCallback(() => {
+    closeDisclaimer();
+  }, [closeDisclaimer]);
+  const onContinue = useCallback(() => {
+    if (disableDisclaimerChecked) {
+      disableDisclaimer();
+    }
+    closeDisclaimer();
+    next();
+  }, [disableDisclaimerChecked, closeDisclaimer, disableDisclaimer, next]);
+
+  return (
+    <BottomModal
+      style={{
+        ...styles.root,
+        backgroundColor: colors.card,
+      }}
+      isOpened={isOpened}
+    >
+      <TouchableOpacity style={styles.closeIcon} onPress={onClose}>
+        <CloseIcon size={18} />
+      </TouchableOpacity>
+
+      <View style={[styles.flowRow, styles.head]}>
+        <View style={[styles.headElements, styles.roundLogoContainer]}>
+          <LiveIcon size={31} color="blue" />
+        </View>
+        {icon && (
+          <>
+            <View style={[styles.headElements, styles.flowRow]}>
+              {Array(6)
+                .fill()
+                .map((_, i) => (
+                  <View
+                    key={i}
+                    style={{ ...styles.dash, backgroundColor: colors.live }}
+                  />
+                ))}
+            </View>
+            <View style={styles.headElements}>
+              <AppIcon size={48} icon={icon} />
+            </View>
+          </>
+        )}
+      </View>
+
+      <LText semiBold style={[styles.textCenter, styles.title]}>
+        <Trans i18nKey={"platform.disclaimer.title"} />
+      </LText>
+      <LText style={[styles.textCenter, styles.description]}>
+        <Trans i18nKey={"platform.disclaimer.description"} />
+      </LText>
+
+      <View
+        style={{
+          ...styles.flowRow,
+          ...styles.blueInfo,
+          backgroundColor: rgba(colors.live, 0.1),
+        }}
+      >
+        <View style={styles.blueInfoIcon}>
+          <InfoIcon color={colors.live} size={18} />
+        </View>
+        <LText style={{ ...styles.blueInfoText, color: colors.live }}>
+          <Trans i18nKey={"platform.disclaimer.legalAdvice"} />
+        </LText>
+      </View>
+
+      <TouchableOpacity
+        style={[styles.flowRow, styles.reminder]}
+        onPress={() => {
+          setDisableDisclaimerChecked(!disableDisclaimerChecked);
+        }}
```

Comment:
Since the new state is computed from the previous state, you could use a [functional update](https://reactjs.org/docs/hooks-reference.html#functional-updates).

```suggestion
        onPress={() => setDisableDisclaimerChecked(state => !state)}
```


Or at least this could be written as a one liner (without curly braces)
```suggestion
        onPress={() => setDisableDisclaimerChecked(!disableDisclaimerChecked)}
```

Reply:
Using the state feels way less intuitive since we already declared the boolean before. I'll change for the one liner 👍